util/raclgen.py

```diff
@@ -39,6 +49,44 @@ def main():
 
     args = parser.parse_args()
 
+    if args.doc:
+        top = _read_hjson(args.doc)
+        if 'racl_ctrl' not in (module['name'] for module in top['module']):
+            raise SystemExit(f'No racl_ctrl module found in {args.doc}')
+        if 'racl' not in top:
+            raise SystemExit(f'Missing key "racl" in {args.doc}')
+
+        racl_config = top['racl']
+        ips_with_racl = {}
+
+        gen_md_header(racl_config)
+
+        for module in top['module']:
+            if 'racl_mappings' not in module:
+                continue
+            ip_name = module['type']
+            if ip_name not in ips_with_racl:
+                try:
+                    ip_path = topgen_lib.get_ip_hjson_path(ip_name, top, REPO_TOP)
+                    ips_with_racl[ip_name] = IpBlock.from_path(path=ip_path, param_defaults=[])
+                except ValueError as err:
+                    raise SystemExit(f'Failed to parse IP block "{ip_name}". Error: {err}')
+
+            for if_name, racl_mappings in module['racl_mappings'].items():
+                racl_mapping = {
+                    'if_name': if_name,
+                    'racl_group': racl_mappings['racl_group'],
```

Comment:
Do we already know that this key is valid for some reason? (An argument for using proper data types!)